src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -179,30 +179,36 @@ public CSVFileScanFragment(
   @Override
   protected TupleBatch fetchNextReady() throws IOException, DbException {
     long lineNumberBegin = lineNumber;
+    boolean unfinishedEncapsulatedTokenRow = false;
 
     while ((buffer.numTuples() < TupleBatch.BATCH_SIZE) && !flagAsIncomplete) {
       lineNumber++;
       if (parser.isClosed()) {
         break;
       }
 
+      if (unfinishedEncapsulatedTokenRow) {
+        onLastRow = true;
+      }
+
       try {
         if (!onLastRow) {
           record = iterator.next();
         }
       } catch (Exception e) {
-        /*
-         * FIX ME: If we hit an exception for a malformed row (in case of quotes for example), we read more bytes by
-         * marking this as the final line
-         */
-        onLastRow = true;
       }
 
       try {
         if (!iterator.hasNext()) {
           onLastRow = true;
         }
       } catch (Exception e) {
```

Comment:
I tried to only catch an ```IOException``` but the compiler complains that it will never be reached. I think this is because it actually throws a RuntimeException? Is it ok if we catch that?